### PR TITLE
Fixes Regex & adds Jest testing & adds Comment Support

### PR DESCRIPTION
**Pull request by TGTGamer**
So firstly, sorry for breaking your action. You probably didn't even notice because I haven't had a message yelling at me yet xD

image

**Features:**
- Adds Jest Support
- Adds Debugging
- Adds Issue Comment Support
- Adds Pull Request Review Support

**Fixes:**
- Fixes Regex issues

## New Debt Created
 String Escape exploration.  The string currently escapes in a way which is weird. If I get the regex string from the `action.yml` or `workflow` it seems to be escaped differently to how strings are handled inside typescript. This needs exploring to ensure that documentation is correct.

Additionally - if you have time, fix it so all strings are the same before they are loaded into the action. The strings in main.test.ts should be the same as the defaults in action.yml

I would have done that, but I'm getting sleepy xD

### CODE REVIEW

**Comment by TGTGamer** [comment]
(suggest you either squash this to make it only 1 commit. I may have made a mess of my commit-tree)

File: action.yml [hunk 1]
```diff
@@ -9,25 +9,29 @@ inputs:
     required: false
     default: '[DEBT]'
   title_start:
-    description: 'Allows you to change the prefix used for debt issues'
+    description: 'Allows you to change the regex used for debt issues'
     required: false
-    default: '<!--\\[DEBT_ISSUE_TITLE\\]-->'
+    default: '<!--\[DEBT_ISSUE_TITLE\]-->'
```
Comment (TGTGamer) [inline action.yml:14]: ## New Debt Created
 String Escape exploration.  The string currently escapes in a way which is weird. If I get the regex string from the `action.yml` or `workflow` it seems to be escaped differently to how strings are handled inside typescript. This needs exploring to ensure that documentation is correct.

Additionally - if you have time, fix it so all strings are the same before they are loaded into the action. The strings in main.test.ts should be the same as the defaults in action.yml

I would have done that, but I'm getting sleepy xD

**Comment by TGTGamer** [comment]
Fairly sure these checks are failing because it's cross-Fork. However, if they continue to fail after merge @me and I'll fix them. 

They are working fine on my version: https://github.com/TGTGamer/debt-collector/pull/25#pullrequestreview-476356224

File: action.yml (hunk 1 above)
Comment (teepsdonnelly) [reply action.yml:14]: No worries! I'm loving all of these changes and appreciate your contributions.

File: __tests__/main.test.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import * as functions from '../src/helpers'
+
+const body =
+  "## New Debt Created \r\n  <!--[DEBT_ISSUE_TITLE]--> Refactor logging in with social accounts. <!--[DEBT_BODY_START]--> Currently, the logic for logging in with social accounts is at the view layer in the app. It needs to be moved out into a helper class and we should utilize the ViewModel appropriately to give the View what it needs. Let's keep our views as dumb as possible for readability and maintainability. \r\n A second paragraph to showcase how an end descriptor can be used too <!--[DEBT_BODY_END]--> Something following it after the end \r\n And another paragraph to show it wont see this"
```
Comment (teepsdonnelly) [inline __tests__/main.test.ts:4]: I like the addition of tests!

**Comment by teepsdonnelly** [comment]
This is all looking great so far!

It looks like the create ticket step is failing -- is that expected on your end or should we fix that before merge?

image

**Comment by teepsdonnelly** [comment]
Ah nevermind, you did mention those above. I'll go ahead and merge now.

File: __tests__/main.test.ts (hunk 2 above)
Comment (TGTGamer) [reply __tests__/main.test.ts:4]: Yeah my head was spinning in circles so I needed to do something to ensure things worked properly 😂

**Pull request merged**